RIOT-OS/RIOT
# doc: move device driver guide to a doxygen page

### Contribution description

- Copied the contents of https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT/a680cbf44276ac8222fd8ae6cb989a66a0720a67 to `DRIVER_GUIDE.md`.
- Some minor modifications without changing any content
    - Enforced 80 chars per line limit where possible
    - Fixed a typo
    - Removed trailing whietspace
    - Added markdown syntax for code highlight in two cases
    - Reduced the TODO section heading from level 1 to level 2 
- Added Doxygen integration
    - Added `[TOC]` to let doxygen create a table of contents (currently shown above right)
    - Added link anchors for consistency with other documentation files
    - Added file to `riot.doxyfile` to include it into the generated doc
   - Used `@code` ... `@endcode` for code blocks. (Fenced code blocks don't parse with the Doxygen version used in the CI.)
- Shorted main header
    - This is needed to properly display in the left menu of the RIOT doc
- Transformed one numbered list containing code blocks as items to regular text
    - Doxygen doesn't support list items containing code blocks

### Testing procedure

1. Verify that the first commit contains a verbatim copy of the maintainer guide
2. Verify that no content is changed with the second commit (except for the minor stuff mentionend above).

### Issues/PRs references

None
